File: QuestionSet/Leetcode/C-Puls-Puls/566. Reshape the Matrix.cpp
/*
 * @Author: FANG
 * @Date: 2021-09-17 17:01:31
 * @LastEditTime: 2021-09-17 17:01:32
 * @Description:Reshape the Matrix
 * @FilePath: \DSA\QuestionSet\Leetcode\C-Puls-Puls\566. Reshape the Matrix.cpp
 */
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        int m = nums.size();
        int n = nums[0].size();
        if (m * n != r * c) return nums;
        
        vector<vector<int>> ans(r, vector<int>(c));
        for (int i = 0; i < m * n; i++) {
            ans[i / c][i % c] = nums[i / n][i % n];
        }
        return ans;
    }
};

File: QuestionSet/Leetcode/Java/566. Reshape the Matrix.java
/*
 * @Author: FANG
 * @Date: 2021-09-17 16:54:44
 * @LastEditTime: 2021-09-17 16:54:44
 * @Description: Reshape the Matrix
 * @FilePath: \DSA\QuestionSet\Leetcode\Java\566. Reshape the Matrix.java
 */
class Solution {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        if (r * c != mat.length * mat[0].length) return mat;
        int rowVal = 0, colVal = 0;
        // int newRowVal = 0, newColVal = 0;
        int[][] newMat = new int[r][c];
        for (int i = 0; i < r; i++) {
            for (int j = 0; j < c; j++) {
                newMat[i][j] = mat[rowVal][colVal];
                colVal++;
                if (colVal % mat[0].length == 0) {
                    rowVal++;
                    colVal = 0;
                }
            }
        }
        return newMat;
    }
}